gems/pending/util/duplicate_blocker/dedup_handler.rb

```diff
@@ -63,8 +68,8 @@ def handle(method, *args)
       key = key_generator.call(method, *args)
       desc = descriptor.call(method, *args)
 
-      entry = update_history(key, desc)
-      tripped?(entry) ? on_blocking_call(desc) : method[*args]
+      entry = update_history(key.hash, desc)
+      tripped?(entry, desc) ? on_blocking_call(desc) : method[*args]
```

Comment:
I do realize that is your bullet point 2, but how much savings are you talking about by removing that.  It does complicate the code quite a bit, for possibly not a lot of gain.

On that note, do you have numbers to back up your changes and how they actually improved things?  Memory before/after/savings% ?